### PR TITLE
OPTI-1411 toll roads with additional weight

### PR DESCRIPTION
**Pull request by kandelirina**

### CODE REVIEW

File: core/src/main/java/com/graphhopper/GraphHopper.java [hunk 1]
```diff
@@ -929,6 +931,12 @@ public Weighting createWeighting(HintsMap hintsMap, FlagEncoder encoder, Graph g
                     parseBlockArea(blockAreaStr, DefaultEdgeFilter.allEdges(encoder), hintsMap.getDouble("block_area.edge_id_max_area", 1000 * 1000));
             return new BlockAreaWeighting(weighting, blockArea);
         }
+        if (addTurnAdditionalWeight) {
+            Collection<Penalty> penalties = new ArrayList<>();
+            penalties.add(new TurnPenalty(hintsMap));
+            weighting = new WeightingWithPenalties(weighting, encoder, hintsMap, penalties);
```
Comment (roir) [inline core/src/main/java/com/graphhopper/GraphHopper.java:937]: here you can override the weighting you assigned in lines 906-922 above, is this on purpose ? if so why not put as an else to the above

File: core/src/main/java/com/graphhopper/routing/weighting/GenericWeightingsWithTollPenalties.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        super(encoder, hintsMap);
+    }
+
+
+    @Override
+    public double calcWeight(EdgeIteratorState edgeState, boolean reverse, int prevOrNextEdgeId) {
+        final Toll toll = tollEnc.getEnum(reverse, edgeState.getFlags());
+        if (toll == Toll.NO)
+            return super.calcWeight(edgeState, reverse, prevOrNextEdgeId);
+        return super.calcWeight(edgeState, reverse, prevOrNextEdgeId)*10;
```
Comment (roir) [inline core/src/main/java/com/graphhopper/routing/weighting/GenericWeightingsWithTollPenalties.java:25]: who is 10?

Comment (kandelirina) [reply core/src/main/java/com/graphhopper/routing/weighting/GenericWeightingsWithTollPenalties.java:25]: oh, deleted this file, sorry

File: core/src/main/java/com/graphhopper/GraphHopper.java (hunk 1 above)
Comment (kandelirina) [reply core/src/main/java/com/graphhopper/GraphHopper.java:937]: on purpose, sorry did not understand you

**Pull request merged**